src/error.js

```diff
@@ -319,16 +319,30 @@ function reportErrorToServer(message, filename, line, col, error) {
   const data = getErrorReportData(message, filename, line, col, error,
       hasNonAmpJs);
   if (data) {
+    reportingBackoff(() => reportErrorToServerOrViewer(this, data));
+  }
+}
+
+/**
+ * Passes the given error data to either server or viewer.
+ * @param {!Window} win
+ * @param {!JsonObject} data Data from `getErrorReportData`.
+ * @return {Promise<undefined>}
+ */
+export function reportErrorToServerOrViewer(win, data) {
+  return new Promise(resolve => {
     // Report the error to viewer if it has the capability. The data passed
     // to the viewer is exactly the same as the data passed to the server
     // below.
-    maybeReportErrorToViewer(this, data);
-    reportingBackoff(() => {
-      const xhr = new XMLHttpRequest();
-      xhr.open('POST', urls.errorReporting, true);
-      xhr.send(JSON.stringify(data));
+    maybeReportErrorToViewer(win, data).then(reportedErrorToViewer => {
```

Comment:
You can just return this promise:

```js
return maybeReportErrorToViewer(win, data).then(reportedErrorToViewer => {
  ...
});
```

That way we don't get dangling promise that's never resolved or rejected when the `maybeReportErrorToViewer` promise chain rejects.

Reply:
Done.